From 30dfdd7b493399405b91623d2cc494a896be6349 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Tue, 1 Aug 2017 13:03:36 +0800
Subject: [PATCH 3036/5242] MLK-15110-16 gpu: imx: dpu: fetcheco: Add helpers
 to set/get fetcheco off pin

commit  056aa94ef33fad53826913d9ef623d383f728d3c from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds some helpers to set/get fetcheco off pin.
We need to pin fetcheco off when the primary plane is disabled and the
relevant fetcheco is feed by prefetch engine.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-fetcheco.c |   19 +++++++++++++++++++
 include/video/dpu.h                |    3 +++
 2 files changed, 22 insertions(+)

diff --git a/drivers/gpu/imx/dpu/dpu-fetcheco.c b/drivers/gpu/imx/dpu/dpu-fetcheco.c
index 28d6e80..91aaec4 100644
--- a/drivers/gpu/imx/dpu/dpu-fetcheco.c
+++ b/drivers/gpu/imx/dpu/dpu-fetcheco.c
@@ -45,6 +45,7 @@ struct dpu_fetcheco {
 	struct mutex mutex;
 	int id;
 	bool inuse;
+	bool pin_off;
 	struct dpu_soc *dpu;
 	/* see DPU_PLANE_SRC_xxx */
 	unsigned int stream_id;
@@ -439,6 +440,24 @@ void fetcheco_set_stream_id(struct dpu_fetcheco *fe, unsigned int id)
 }
 EXPORT_SYMBOL_GPL(fetcheco_set_stream_id);
 
+void fetcheco_pin_off(struct dpu_fetcheco *fe)
+{
+	fe->pin_off = true;
+}
+EXPORT_SYMBOL_GPL(fetcheco_pin_off);
+
+void fetcheco_unpin_off(struct dpu_fetcheco *fe)
+{
+	fe->pin_off = false;
+}
+EXPORT_SYMBOL_GPL(fetcheco_unpin_off);
+
+bool fetcheco_is_pinned_off(struct dpu_fetcheco *fe)
+{
+	return fe->pin_off;
+}
+EXPORT_SYMBOL_GPL(fetcheco_is_pinned_off);
+
 struct dpu_fetcheco *dpu_fe_get(struct dpu_soc *dpu, int id)
 {
 	struct dpu_fetcheco *fe;
diff --git a/include/video/dpu.h b/include/video/dpu.h
index 45d253e..9ee7427 100644
--- a/include/video/dpu.h
+++ b/include/video/dpu.h
@@ -563,6 +563,9 @@ void fetcheco_frameresampling(struct dpu_fetcheco *fe, unsigned int x,
 dpu_block_id_t fetcheco_get_block_id(struct dpu_fetcheco *fe);
 unsigned int fetcheco_get_stream_id(struct dpu_fetcheco *fe);
 void fetcheco_set_stream_id(struct dpu_fetcheco *fe, unsigned int id);
+void fetcheco_pin_off(struct dpu_fetcheco *fe);
+void fetcheco_unpin_off(struct dpu_fetcheco *fe);
+bool fetcheco_is_pinned_off(struct dpu_fetcheco *fe);
 struct dpu_fetcheco *dpu_fe_get(struct dpu_soc *dpu, int id);
 void dpu_fe_put(struct dpu_fetcheco *fe);
 
-- 
1.7.9.5

